google/cloud/aiplatform/_featurestores/entity_type.py

```diff
@@ -0,0 +1,456 @@
+                or "my_entity_type_id" when project and location are initialized or passed, with featurestore_id passed.
+            featurestore_id (str):
+                Optional. Featurestore to retrieve entityType from.
+            project (str):
+                Optional. Project to retrieve entityType from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve entityType from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to retrieve this EntityType. Overrides
+                credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the provided entity_type_name is not in form of a fully-qualified
+            entityType resource name nor an entity_type ID with featurestore_id passed.
+        """
+
+        match = featurestore_utils.validate_entity_type_name(entity_type_name)
+
+        if match:
+            self._featurestore_id = match["featurestore_id"]
+        elif (
+            featurestore_utils.validate_id(entity_type_name)
+            and featurestore_id
+            and featurestore_utils.validate_id(featurestore_id)
+        ):
+            self._featurestore_id = featurestore_id
+        else:
+            raise ValueError(
+                f"{entity_type_name} is not in form of a fully-qualified entityType resource name "
+                f"nor an entity_type ID with featurestore_id passed."
+            )
+
+        self._resource_noun = featurestore_utils.get_entity_type_resource_noun(
+            featurestore_id=self._featurestore_id
+        )
+
+        super().__init__(
+            project=project,
+            location=location,
+            credentials=credentials,
+            resource_name=entity_type_name,
+        )
+        self._gca_resource = self._get_gca_resource(resource_name=entity_type_name)
+
+        self._featurestore_name = utils.full_resource_name(
+            resource_name=self._featurestore_id,
+            resource_noun="featurestores",
+            project=self.project,
+            location=self.location,
+        )
+
+    @property
+    def featurestore_name(self) -> str:
+        """Full qualified resource name of the managed featurestore in which this EntityType is."""
+        return self._featurestore_name
+
+    def get_featurestore(self) -> _featurestores.Featurestore:
+        """Retrieves the managed featurestore in which this EntityType is.
+
+        Returns:
+            featurestores.Featurestore - The managed featurestore in which this EntityType is.
+        """
+        return _featurestores.Featurestore(self._featurestore_name)
+
+    def get_feature(self, feature_id: str) -> "_featurestores.Feature":
```

Comment:
Do we expect the feature resource exist (i.e it was created before)? If not, then should the (Feature) update function throw error in that case?

Similar for get_entity_type for featurestore class.

Reply:
Both get_* methods will pass in the fully-qualified resource name into resource class constructor. If the resource exists, the method will return the constructed resource class, if the resource does not exist, it will actually hit the service client's get_* method and throw service error.